beacon-chain/operations/voluntaryexits/service.go

```diff
@@ -57,60 +58,58 @@ func (p *Pool) PendingExits(state *beaconstate.BeaconState, slot uint64, noLimit
 }
 
 // InsertVoluntaryExit into the pool. This method is a no-op if the pending exit already exists,
-// has been included recently, or the validator is already exited.
+// or the validator is already exited.
 func (p *Pool) InsertVoluntaryExit(ctx context.Context, state *beaconstate.BeaconState, exit *ethpb.SignedVoluntaryExit) {
 	ctx, span := trace.StartSpan(ctx, "exitPool.InsertVoluntaryExit")
 	defer span.End()
-
 	p.lock.Lock()
 	defer p.lock.Unlock()
 
-	// Has this validator index been included recently?
-	if p.included[exit.Exit.ValidatorIndex] {
+	// Prevent malformed messages from being inserted.
+	if exit == nil || exit.Exit == nil {
 		return
 	}
 
-	// Has the validator been exited already?
-	if v, err := state.ValidatorAtIndexReadOnly(exit.Exit.ValidatorIndex); err != nil || v.ExitEpoch() != params.BeaconConfig().FarFutureEpoch {
+	// Prevent any sort of duplicate exit from being inserted.
+	existsInPending, _ := existsInList(p.pending, exit.Exit.ValidatorIndex)
```

Comment:
I think we still need to handle this case. 
```go
		// If an exit exists with this validator index, prefer one with an earlier exit epoch.
```
In the current case we will reject it outright instead of favouring the one with a lower exit epoch.